src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergTableSchemaValidatorIT.java

```diff
@@ -33,44 +34,115 @@ public void tearDown() {
     dropIcebergTable(tableName);
   }
 
-  @Test
-  public void shouldValidateExpectedIcebergTableSchema() {
-    // given
-    createIcebergTable(tableName);
-    enableSchemaEvolution(tableName);
+  @Nested
+  class SchemaEvolutionEnabled {
+    public static final boolean SCHEMA_EVOLUTION = true;
 
-    // when, then
-    schemaValidator.validateTable(tableName, TEST_ROLE);
-  }
+    @Test
+    public void shouldValidateExpectedIcebergTableSchema() {
+      // given
+      createIcebergTable(tableName);
+      enableSchemaEvolution(tableName);
 
-  @Test
-  public void shouldThrowExceptionWhenTableDoesNotExist() {
-    Assertions.assertThrows(
-        SnowflakeKafkaConnectorException.class,
-        () -> schemaValidator.validateTable(tableName, TEST_ROLE));
-  }
+      // when, then
+      schemaValidator.validateTable(tableName, TEST_ROLE, SCHEMA_EVOLUTION);
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenTableDoesNotExist() {
```

Comment:
`@Nested` is cool, but I think it would be more concise in this particular example:
```
@ParameterizedTest
  @ValueSource(booleans = {true, false})
  public void shouldThrowExceptionWhenTableDoesNotExist(boolean schemaEvolution) {
    Assertions.assertThrows(
            SnowflakeKafkaConnectorException.class,
            () -> schemaValidator.validateTable(tableName, TEST_ROLE, schemaEvolution));
  }
```